typescript/web/src/components/pagination-tool/pagination-tool.tsx

```diff
@@ -0,0 +1,134 @@
+  Text,
+} from "@chakra-ui/react";
+import { RiArrowLeftSLine, RiArrowRightSLine } from "react-icons/ri";
+import { isNaN, clamp } from "lodash/fp";
+import { useEffect, useState } from "react";
+
+interface PaginationToolProps {
+  currentPage: number;
+  maxPage: number;
+  onChange: (newPage: number) => void;
+}
+
+const digitsPerRem = 0.55;
+
+export const PaginationTool = (props: PaginationToolProps) => {
+  const { currentPage, maxPage, onChange: setPage } = props;
+  const [inputValue, setInputValue] = useState(currentPage);
+  const digitCount = Math.max(1, Math.ceil(Math.log10(maxPage ?? 1)));
+
+  useEffect(() => {
+    setInputValue(currentPage);
+  }, [currentPage]);
+
+  const handleKeyPress = (event: any) => {
+    if (event.key === "Enter") {
+      setPage(event.target.value as number);
+    }
+    if (event.key === "Escape" || event.key === "Esc") {
+      event.preventDefault();
+      setInputValue(currentPage);
+    }
+  };
+
+  return (
+    <HStack h={10} p={0} spacing={1} rounded={6} pointerEvents="initial">
+      {currentPage > 1 ? (
+        <Tooltip
+          openDelay={300}
+          label={`Previous Page: ${currentPage - 1}`}
+          placement="top"
+        >
+          <IconButton
+            bg="none"
+            aria-label="Previous page"
+            icon={<RiArrowLeftSLine size="1.5em" />}
+            onClick={() => setPage(currentPage - 1)}
+          />
+        </Tooltip>
+      ) : (
+        <IconButton
+          disabled
+          bg="none"
+          aria-label="No previous page"
+          icon={<RiArrowLeftSLine size="1.5em" />}
+        />
+      )}
+      <Tooltip label="Current page" placement="top" openDelay={300}>
+        <NumberInput
+          name="current-page"
+          rounded={6}
+          allowMouseWheel
+          defaultValue={currentPage != null ? currentPage : 1}
+          min={1}
+          max={maxPage}
+          variant="filled"
+          textAlign="center"
+          size="sm"
+          value={inputValue}
+          onChange={(valStr, valNum) => {
+            if (!isNaN(valNum)) {
+              setInputValue(clamp(valNum, 1, maxPage));
+            }
+          }}
+          onKeyPress={handleKeyPress}
+          onBlur={() => {
+            setPage(inputValue);
+          }}
+        >
+          <NumberInputField
+            rounded={6}
+            onClick={(event) => {
+              (event.target as HTMLInputElement).select();
+            }}
+            textAlign="center"
+            w={`${digitCount * digitsPerRem + 1.5}rem`}
+            pr={2}
+            pl={1.5}
+          />
+        </NumberInput>
+      </Tooltip>
+      <Text textAlign="center" userSelect="none" pl={2}>
+        /
+      </Text>
+      <Text
+        mr={3}
+        ml={2}
+        w={`${digitCount * digitsPerRem + 0.5}rem`}
+        textAlign="left"
+        cursor="default"
+        fontSize="sm"
+      >
+        {maxPage}
+      </Text>
+      {currentPage < maxPage ? (
```

Comment:
Same thing as before, I'd avoid the conditional rendering by using the `isDisabled` prop of the Tooltip component